pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -160,55 +162,54 @@ func (exp *flowExporter) Export(collector net.Addr, stopCh <-chan struct{}, poll
 
 }
 
-func (exp *flowExporter) initFlowExporter(collector net.Addr) error {
+func (exp *flowExporter) initFlowExporter(collectorAddr string, collectorProto string) error {
 	// Create IPFIX exporting expProcess, initialize registries and other related entities
 	obsID, err := genObservationID()
 	if err != nil {
 		return fmt.Errorf("cannot generate obsID for IPFIX ipfixexport: %v", err)
 	}
 
-	// To connect to the flow aggregator, we send collector in the format, where
-	// IP of the host is empty string.
 	// We use the static DNS name to connect to the flow aggregator.
-	host, _, _ := net.SplitHostPort(collector.String())
-	if host == "" {
+	if strings.Contains(collectorAddr, FlowAggregatorDNSName) {
 		hostIPs, err := net.LookupIP(FlowAggregatorDNSName)
 		if err != nil {
 			return err
 		}
+		// Currently, supporting only IPv4 for Flow Aggregator.
 		ip := hostIPs[0].To4()
 		if ip != nil {
-			// Re-initialize the collector address with resolved flow aggregator
-			// host IP.
-			if collector.Network() == "tcp" {
-				collector = &net.TCPAddr{
-					IP:   ip,
-					Port: 4739,
-					Zone: "",
-				}
-			} else {
-				collector = &net.UDPAddr{
-					IP:   ip,
-					Port: 4739,
-					Zone: "",
-				}
-			}
+			// Update the collector address with resolved IP of flow aggregator
+			collectorAddr = net.JoinHostPort(ip.String(), defaultIPFIXPort)
```

Comment:
do we need to resolve the IP here? I know that we can directly pass the dns name into collector and it will automatically resolve it. 
Also, passing the dns name will help verify TLS cert more easily because we can pass DNS name when creating the cert rather than resolved IP.

Reply:
I tried that in the beginning using the FQDN name in netAddr. It did not work for some reason. Let me give it one more shot.

Reply:
The restriction is that go-ipfix exporting process requires an input in net.Addr while taking collector input. net.Addr can be acquired through either net.ResolveTCPAddr or net.TCPAddr, and similarly for UDP. If we could remove that restriction and accept the string as mentioned in the code, then we could do that.

Reply:
Got it. Thanks for clarifying it.